### PR TITLE
Add Python wrap of conv2d_transpose and its unittest

### LINKED ISSUE #5965: Python conv2d_transpose layer for fluid
It is a crucial layer for DCGAN.

### PR DESCRIPTION
**Pull request by reyoung**
conv2d_transpose is a crucial layer for DCGAN.

Fix #5965

### CODE REVIEW

File: python/paddle/v2/fluid/layers.py [hunk 1]
```diff
@@ -1587,6 +1587,97 @@ def array_length(array, main_program=None):
     return tmp
 
 
+def conv2d_transpose(input,
+                     num_filter,
```
Comment (qingqing01) [inline python/paddle/v2/fluid/layers.py]: https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/fluid/layers.py#L704 

`conv2d_transpose`  和 `conv2d` 里的num_filter单复数保持一致吧。

File: python/paddle/v2/fluid/layers.py [hunk 2]
```diff
@@ -1587,6 +1587,97 @@ def array_length(array, main_program=None):
...
+                     output_size=None,
+                     filter_size=None,
+                     padding=None,
+                     stride=None,
+                     param_attr=None,
+                     param_initializer=None,
+                     main_program=None,
+                     startup_program=None):
+    """
+    The transpose of conv2d layer.
```
Comment (qingqing01) [inline python/paddle/v2/fluid/layers.py]: The transposed conv2d layer.

File: python/paddle/v2/fluid/layers.py [hunk 3]
```diff
@@ -1587,6 +1587,97 @@ def array_length(array, main_program=None):
...
+    
+    This layer is also known as deconvolution layer.
+    
+    Args:
+        input(Variable): The input image. [N, C, H, W].
```
Comment (qingqing01) [inline python/paddle/v2/fluid/layers.py]: The input image with [N, C, H, W] format.

File: python/paddle/v2/fluid/tests/test_layers.py [hunk 4]
```diff
@@ -65,6 +65,15 @@ def test_simple_conv2d(self):
 
         print str(program)
 
+    def test_deconv2d(self):
```
Comment (chengduoZH) [inline python/paddle/v2/fluid/tests/test_layers.py]: `test_deconv2d `->`test_conv2d_transpose`

File: python/paddle/v2/fluid/layers.py (hunk 1 above)
Comment (chengduoZH) [inline python/paddle/v2/fluid/layers.py]: `num_filter`改成`output_channel`怎么样？这样更直观一些

File: python/paddle/v2/fluid/layers.py [hunk 5]
```diff
@@ -1587,6 +1587,97 @@ def array_length(array, main_program=None):
     return tmp
 
 
+def conv2d_transpose(input,
+                     num_filter,
+                     output_size=None,
```
Comment (chengduoZH) [inline python/paddle/v2/fluid/layers.py]: 调用`conv2d_transpose`的时候很少用指定`output_size`吧。
另外`padding`, `stride`的默认值（0，1）直接写在函数参数里面是不是更好一些，而不是设为None，因为`padding = 0, stride=1`这样更直观，用户不用再看函数里面到底是怎么设置，我看pytorch、mxnet都是这么做的

Comment (reyoung) [reply python/paddle/v2/fluid/layers.py]: output_size is a very useful argument for GAN.

Users can set either filter_size or output_size

File: python/paddle/v2/fluid/layers.py (hunk 1 above)
Comment (reyoung) [reply python/paddle/v2/fluid/layers.py]: API一致性更重要

File: python/paddle/v2/fluid/tests/test_layers.py (hunk 4 above)
Comment (reyoung) [reply python/paddle/v2/fluid/tests/test_layers.py]: Done

File: python/paddle/v2/fluid/layers.py (hunk 3 above)
Comment (reyoung) [reply python/paddle/v2/fluid/layers.py]: Done

File: python/paddle/v2/fluid/layers.py (hunk 1 above)
Comment (reyoung) [reply python/paddle/v2/fluid/layers.py]: Done.

**Pull request merged**